### PR TITLE
Add Expr.ofFunction1

### PR DESCRIPTION
**Pull request by Atry**

### CODE REVIEW

File: library/src/scala/quoted/Expr.scala [hunk 1]
```diff
@@ -52,6 +52,16 @@ object Expr {
   def unapply[T](x: Expr[T])(using FromExpr[T])(using Quotes): Option[T] =
     scala.Predef.summon[FromExpr[T]].unapply(x)
 
+  /** Creates an expression that will construct a function with one parameter
+   *
+   *  Transforms a sequence of expression
```
Comment (dwijnand) [inline library/src/scala/quoted/Expr.scala:57]:
```suggestion
   *  Transforms the expression
```

**Comment by dwijnand** [comment]
I think we should mark this as `@experimental`, that way we can decide later if we want to commit to having it.

File: library/src/scala/quoted/Expr.scala [hunk 2]
```diff
@@ -52,6 +52,16 @@ object Expr {
...
     scala.Predef.summon[FromExpr[T]].unapply(x)
 
+  /** Creates an expression that will construct a function with one parameter
+   *
+   *  Transforms the expression
+   *    `a => r` where `a: Expr[A]` and `r: Expr[R]`
+   *  to an expression equivalent to
+   *    `'{ $a => $r }` typed as an `Expr[A => R]`
+   */
+  def ofFunction1[A, R](f: Expr[A] => Expr[R])(using Type[R], Type[A])(using Quotes): Expr[A => R] =
```
Comment (nicolasstucki) [inline library/src/scala/quoted/Expr.scala:62]: If we add this we should add it for all function arities. It seems that we would need `TupledFunction` (see #10416) to be able to write this interface for all function sizes.

File: library/src/scala/quoted/Expr.scala [hunk 3]
```diff
@@ -52,6 +52,16 @@ object Expr {
   def unapply[T](x: Expr[T])(using FromExpr[T])(using Quotes): Option[T] =
     scala.Predef.summon[FromExpr[T]].unapply(x)
 
+  /** Creates an expression that will construct a function with one parameter
+   *
+   *  Transforms the expression
+   *    `a => r` where `a: Expr[A]` and `r: Expr[R]`
+   *  to an expression equivalent to
+   *    `'{ $a => $r }` typed as an `Expr[A => R]`
```
Comment (prolativ) [inline library/src/scala/quoted/Expr.scala:60]: As you're adding a new method to stdlib's API you should mark it as`@experimental`

**Comment by nicolasstucki** [comment]
We will not be able to advance on this before we have #14414. I will close this PR for now. We should revisit it once we have #14414.

**Pull request closed without merging**